Document PostProcessorRegistration (#149)
google/libprotobuf-mutator: README.md

Please see [libfuzzer_example.cc](/examples/libfuzzer/libfuzzer_example.cc) as an example.

### Mutation post-processing (experimental)

> Sometimes it's neccecary to keep particular values in some fields whithout which the proto

Reviewer: ```suggestion
Sometimes it's necessary to keep particular values in some fields without which the proto
```

is going to be rejected by fuzzed code. E.g. code may expect consistency between some fields

> or it may use some fields as checksums. Such constrains are going to be significant bottleneck

Reviewer: ```suggestion
or it may use some fields as checksums. Such constraints are going to be significant bottleneck
```

> for fuzzer even if it's capabale to get acceptable values with time.

Reviewer: ```suggestion
for fuzzer even if it's capable of inserting acceptable values with time.
```

PostProcessorRegistration can be used to avoid such issue and guide your fuzzer towards interesing
code. It registers callback which will be called for each message of particular type after each mutation.

```
DEFINE_PROTO_FUZZER(const MyMessageType& input) {
  static PostProcessorRegistration reg = {

>       [](const MyMessageType* message, unsigned int seed) {

Reviewer: This is not `const` right? Because we change it in place?

        TweakMyMessageType(message, seed);
      }};

  // Code which needs to be fuzzed.
  ConsumeMyMessageType(input);
}
```

> Optional: Use seed if callback uses random numbers. It may help later with debuggin.

Reviewer: ```suggestion
Optional: Use seed if callback uses random numbers. It may help later with debugging.
```

Author: Can you add another email into CLA?
https://github.corp.google.com/prinfo?q=/https://github.com/google/libprotobuf-mutator/pull/149
